src/Microsoft.Health.Dicom.Core/Features/Indexing/InstanceReindexer.cs

```diff
@@ -0,0 +1,23 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Indexing
+{
+    /// <summary>
+    /// Reindexer which reindexes DICOM instance.
+    /// </summary>
+    public class InstanceReindexer : IInstanceReindexer
+    {
+        public Task ReindexInstanceAsync(IReadOnlyList<ExtendedQueryTagStoreEntry> entries, long watermark, CancellationToken cancellationToken = default)
```

Comment:
QueueReindexAsync()?

Reply:
Currently we don't Queue tags for reindexing, instead simple have [HttpTrigger ](https://github.com/microsoft/dicom-server/blob/15315868594b0df61e3d9d345b0185ad3e917748/src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Trigger.cs#L65) which add tags to database then start reindexing.